crypto/threads_pthread.c

```diff
@@ -89,8 +94,10 @@ int CRYPTO_THREAD_write_lock(CRYPTO_RWLOCK *lock)
     if (pthread_rwlock_wrlock(lock) != 0)
         return 0;
 # else
-    if (pthread_mutex_lock(lock) != 0)
+    if (pthread_mutex_lock(lock) != 0) {
+        assert(errno != EDEADLK && errno != EBUSY);
```

Comment:
You probably want to change CRYPTO_THREAD_unlock() too, and check for EPERM there

Reply:
> You probably want to change CRYPTO_THREAD_unlock() too, and check for EPERM there

fixup commit pushed; thanks.